#!/usr/bin/env bash

bash run_pipeline_BICRO28.sh
bash run_pipeline_BICRO68.sh
bash run_pipeline_BICRO36.sh
bash run_pipeline_BICRO69.sh
bash run_pipeline_BICRO21.sh
bash run_pipeline_BICRO67.sh
bash run_pipeline_BICRO50.sh
bash run_pipeline_BICRO70.sh
bash run_pipeline_BICRO43.sh
bash run_pipeline_BICRO44.sh
bash run_pipeline_BICRO52.sh
bash run_pipeline_BICRO48.sh
bash run_pipeline_BICRO97.sh
####################################
# bash run_pipeline_BICRO100.sh
# bash run_pipeline_BICRO75.sh
# bash run_pipeline_BICRO76.sh
# bash run_pipeline_BICRO77.sh
# bash run_pipeline_BICRO84.sh
# bash run_pipeline_BICRO97.sh

###################################################
# EXP=XZ68
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ69
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ70
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ71
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ72
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ64
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO50/FASTQ/XZ64_S2_LALL_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/TOBEDONE! $cutsite $fastq 

# EXP=XZ74
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ75
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ76
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ77
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ78
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq 

# EXP=XZ33
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# time bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ37
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# time bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ31
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ38
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ43
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver-seq/BICRO43/fastq/XZ43_S1_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ45
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/home/garner1/Work/dataset/reduced_sequencing/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ63
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO48/FASTQ/XZ63_S2_LALL_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq 

# EXP=XZ64
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO50/FASTQ/XZ64_S2_LALL_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq 

##############################################
# EXP=XZ31
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO67/FASTQ/"$EXP".fastq.gz
# time bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ33
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO68/FASTQ/"$EXP".fastq.gz
# time bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

###############################################################################

# EXP=XZ78
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO65/FASTQ/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq 

# EXP=XZ74
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO65/FASTQ/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ75
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO65/FASTQ/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ76
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO65/FASTQ/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ77
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO65/FASTQ/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq
###############################################################################

# EXP=XZ31
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver-seq/BICRO21/XZ31/XZ31_S2_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ33
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver-seq/BICRO28/XZ33/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ37
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver-seq/BICRO36/fastq/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ38
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO50/FASTQ/XZ38_S2_LALL_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ43
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver-seq/BICRO43/fastq/XZ43_S1_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ45_BICRO44
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver-seq/BICRO44/fastq/XZ45_S1_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ45_BICRO52
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO52/XZ45_S2_LALL_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ63
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO48/FASTQ/XZ63_S2_LALL_R1_001.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq 


################################################################################
# EXP=XZ82
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO57/FASTQ/"$EXP"*/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ83
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO57/FASTQ/"$EXP"*/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ85
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO57/FASTQ/"$EXP"*/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ86
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO57/FASTQ/"$EXP"*/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ88
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO57/FASTQ/"$EXP"*/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ89
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO57/FASTQ/"$EXP"*/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ90
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO57/FASTQ/"$EXP"*/"$EXP".fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

###########################################################################
# EXP=XZ68
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ69
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ70
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ71
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ72
# genome=human
# mode=SE
# cutsite=AAGCTT
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ74
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ75
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ76
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ77
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

# EXP=XZ78
# genome=human
# mode=SE
# cutsite=CATG
# fastq=/media/bicroserver_2-seq/BICRO56/FASTQ/"$EXP"*/"$EXP"_R1.fastq.gz
# bash main.sh $EXP $genome $mode ../pattern/barcode-cutsite_"$EXP" $cutsite $fastq

########################################################################################
# pattern=/home/garner1/Work/pipelines/restseq/pattern
# data=/home/garner1/Work/dataset/restseq

# exp=NC53; 
# RUN=AVO41; MODE=PE; CUTSITE=AAGCTT; 
# # cat /media/bicroserver-seq/Hubrecht/"$RUN"/"$exp"*R1*.fastq.gz > "$data"/"$exp"_"$RUN"_R1.fastq.gz;
# # cat /media/bicroserver-seq/Hubrecht/"$RUN"/"$exp"*R2*.fastq.gz > "$data"/"$exp"_"$RUN"_R2.fastq.gz;
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2 "$data"/"$exp"_"$RUN"_R?.fastq.gz;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC55; 
# RUN=AVO41; MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2 "$data"/"$exp"_"$RUN"_R?.fastq.gz;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC56;
# RUN=AVO41; MODE=SE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2 "$data"/"$exp"_"$RUN"_R?.fastq.gz;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC57;
# RUN=AVO41; MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2 "$data"/"$exp"_"$RUN"_R?.fastq.gz;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC421;
# RUN=AVO33; MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC432;
# RUN=AVO33; MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC6466;
# MODE=PE; CUTSITE=GAATTC; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ1;
# MODE=PE; CUTSITE=GAATTC; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ4;
# MODE=SE; CUTSITE=GAATTC; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# # #############################################################################################################

# exp=NC24; 
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC25;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC26;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC87;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ13;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ14;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ16;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ17;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ18;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ19;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ20;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ21;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ22;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ23;
# MODE=SE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ24;
# MODE=SE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=XZ30;
# MODE=PE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=BICRO22_XZ16;
# MODE=SE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_XZ16 "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=BICRO22_XZ17;
# MODE=SE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_XZ17 "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=BICRO22_XZ32;
# MODE=SE; CUTSITE=AAGCTT; 
# # ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_XZ32 "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC88;
# MODE=PE; CUTSITE=AAGCTT; 
# ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC89;
# MODE=PE; CUTSITE=AAGCTT; 
# ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC101;
# MODE=PE; CUTSITE=AAGCTT; 
# ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

# exp=NC105;
# MODE=PE; CUTSITE=AAGCTT; 
# ./main.sh "$exp" human 1 10000000 "$MODE" ~/Work/pipelines/restseq/pattern/barcode-cutsite_"$exp" "$CUTSITE" 2;
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_R.dat & pid1=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "+"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_R.dat & pid2=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 == 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__eq0_L.dat & pid3=$!
# cat "$data"/"$exp"/outdata/cutsite_dist_strand_qScore_UMI_ID_start__*_q1.bed | awk '$4 != 0 && $5 == "-"' | wc -l > "$data"/"$exp"/outdata/enzyme_specificity__neq0_L.dat & pid4=$!
# wait $pid1
# wait $pid2
# wait $pid3
# wait $pid4

























